arango/cursors/base.py

```diff
@@ -5,25 +5,30 @@
     CursorNextError,
     CursorCloseError,
 )
+from arango import APIWrapper
+from arango import Request
+from arango.jobs import BaseJob
 
 
-class Cursor(object):
+class BaseCursor(APIWrapper):
     """ArangoDB cursor which returns documents from the server in batches.
 
     :param connection: ArangoDB database connection
-    :type connection: arango.connection.Connection
+    :type connection: arango.connections.BaseConnection
     :param init_data: the cursor initialization data
     :type init_data: dict
+    :param cursor_type: One of `"cursor"` or `"export"`, which endpoint to use
     :raises CursorNextError: if the next batch cannot be retrieved
     :raises CursorCloseError: if the cursor cannot be closed
 
     .. note::
         This class is designed to be instantiated internally only.
     """
 
-    def __init__(self, connection, init_data):
-        self._conn = connection
+    def __init__(self, connection, init_data, cursor_type="cursor"):
```

Comment:
I see that for jobs we pass in the class, but for cursor types we pass in magic string. This seems inconsistent. Also, why do we have to pass in such information using a new argument? Have you considered using `isinstance`?

Reply:
The string itself used in the format string is the only api difference between the two cursors, they otherwise behave identically on the client side. We could create a wrapper class, but that adds a lot of extra complexity for a single string change

Reply:
Doesn't this also break backwards compatibility? `Cursor` and `ExportCursor` should remain available to the users.

Reply:
Okay, my first comment was incorrect, I forgot I had kept the ExportCursor class.  I'll use a setter instead of having this in the init.